official/resnet/resnet_run_loop.py

```diff
@@ -168,8 +168,14 @@ def learning_rate_with_decay(
   vals = [initial_learning_rate * decay for decay in decay_rates]
 
   def learning_rate_fn(global_step):
+    """Builds scaled learning rate function with 5 epoch warm up."""
     global_step = tf.cast(global_step, tf.int32)
-    return tf.train.piecewise_constant(global_step, boundaries, vals)
+    lr = tf.train.piecewise_constant(global_step, boundaries, vals)
+    warmup_steps = int(batches_per_epoch * 5)
```

Comment:
Similarly, I think the cast to int here should be unnecessary? Unlikely to matter for performance, but makes things simpler.